### PR TITLE
feat: add sticky table header option

### LINKED ISSUE #728: Add option to make table header sticky

### PR DESCRIPTION
**Pull request by moonayyur**
closes #728

### CODE REVIEW

**Review by stropitek — COMMENTED** [review]
Why didn't you put the sticky option in the `Header` component?


CleanShot 2024-05-27 at 10 33 04

We shouldn't be able to see the table rows which are above the header.

**Comment by moonayyur** [comment]
> Why didn't you put the sticky option in the `Header` component?

Should I use `cloneElement` to pass the prop to `Header` then ?

**Comment by stropitek** [comment]
> Should I use `cloneElement` to pass the prop to `Header` then ?

Can you explain? I don't think you need to do that in order to move the sticky prop from the `Table` component to the `TableHeader` component.

File: src/components/table/Table.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ export interface TableProps extends React.HTMLAttributes<HTMLTableElement> {
   compact?: boolean;
   interactive?: boolean;
   striped?: boolean;
+  stickyHeader?: boolean;
```
Comment (stropitek) [inline src/components/table/Table.tsx:37]: This should only be in the props of the `TableHeader` component and be called `sticky`.

**Review by stropitek — APPROVED** [review]

**Pull request merged**